### ISSUE TITLE
False positive: [EA] Unnecessary type casting

### ISSUE THREAD

**Issue by Bilge** [kalessil/phpinspectionsea#1980]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) |
| Language level | e.g. PHP 8.4|

Current behaviour
--
```php
$foo = [];
(bool)current($foo); // EA Unnecessary type casting
```

Expected behaviour
--

`current` returns `false|mixed`. In this declaration (`false|mixed`), `false` is redundant because it is masked by `mixed`. Since it can be anything at all, the cast is not unnecessary at all.
